arangod/V8Server/v8-collection.cpp

```diff
@@ -191,15 +227,12 @@ static int ParseDocumentOrDocumentHandle(v8::Isolate* isolate,
     if (ServerState::instance()->isCoordinator()) {
       ClusterInfo* ci = ClusterInfo::instance();
       try {
-        std::shared_ptr<LogicalCollection> col =
-            ci->getCollection(vocbase->name(), collectionName);
-        auto colCopy = col->clone();
-        collection = colCopy.release();
+        collection = ci->getCollection(vocbase->name(), collectionName).get();
```

Comment:
this seems a bit dangerous, shouldn't you be keeping the shared_ptr copy ?

Reply:
done